torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -22,9 +23,12 @@ void serialize(const Message& message, std::ostream& os) {
   std::vector<torch::Tensor> tensors = message.tensors();
   // append payload as a tensor
   tensors.push_back(torch::from_blob(payload, payload_size, {torch::kChar}));
-  // append id as a tensor
-  tensors.push_back(torch::tensor({message.id()}, {torch::kInt64}
-  ));
+  // append id and autograd metadata as a tensor
+  tensors.push_back(torch::tensor(
```

Comment:
If there are dedicated message types for distributed autograd, we might be able to avoid making changes here. The changes to the recv task can be added to the `RequestCallback` implementations I think.